### PR TITLE
Refactor ContentId: UUIDv4 -> randomBytes + encodeAddress

### PR DESCRIPTION
**Pull request by siman**
Fix https://github.com/Joystream/apps/issues/149

### CODE REVIEW

File: packages/joy-media/src/types.ts [hunk 1]
```diff
@@ -2,9 +2,56 @@ import { Enum, Struct, Option, Vector } from '@polkadot/types/codec';
...
+export class ContentId extends Hash {
+
+  static generate (): ContentId {
+    // randomAsU8a uses https://www.npmjs.com/package/tweetnacl#random-bytes-generation
+    return new ContentId(randomAsU8a());
+  }
+
+  /** This function is for backward-compatibility with content ids that were generated as UUID. */
+  static isUuidFormat (contentId: string | Uint8Array): boolean {
+    let hexU8: Uint8Array =
```
Comment (jfinkhaeuser) [inline packages/joy-media/src/types.ts]: Since you actually encoded some of the `-` characters into the string saved in substrate, this function could be as easy as finding dashes. Base58 would not include them.

This only works if you're still storing the encoded addresses in the runtime, which is obviously not part of this PR here.

Comment (siman) [reply packages/joy-media/src/types.ts]: Makes sense! Will use approach with dashes.

Comment (siman) [reply packages/joy-media/src/types.ts]: Refactored here: https://github.com/Joystream/apps/pull/158/commits/d6e512ba31e59df93830ac03c8bdd4dfd6b5df44

**Comment by mnaamani** [comment]
Something occurred to me right after merging..

The address encoding will have a prefix (by default 42 if not specified or not specified by the chain configuration which is discovered with an RPC call to the node we are communicating with) 

https://github.com/Joystream/apps/blob/15c47575111779c836d681c94787273b007acb01/packages/ui-api/src/Api.tsx#L113

Since we might choose to change this value in future to be unique to joystream the encoded contentId will not be "portable". (If we ever want to export chain state and reuse it on a new chain and preserve storage node repositories content)
